### ISSUE TITLE
Pull up to refresh ?

### ISSUE THREAD

**Issue by farzadi** [dkzwm/SmoothRefreshLayout#27]
I wanna set refresh to the bottom how can I do this?and how can I change the style to circular color progress bar.
Thanks.

**Comment by dkzwm** [comment]
Do you need to use the Footer provided by the library or customize your own Footer?
There are material Footer in the library，you can see this class to know how to customize your own Footer.
If you wanna set footer refresh(LoadMore) provided by the library, You can set
```
mRefreshLayout = (SmoothRefreshLayout) findViewById(R.id.refreshLayout);
mRefreshLayout.setDisableLoadMore(false);
MaterialFooter footer=new MaterialFooter(context);
footer.setProgressBarColors(new int[]{Color.RED, Color.BLUE, Color.GREEN, Color.BLACK});
mRefreshLayout.setFooterView(footer);
mRefreshLayout.setDisableRefresh(true);
mRefreshLayout.setOnRefreshListener(new RefreshingListenerAdapter() {
            @Override
            public void onRefreshBegin(final boolean isRefresh //In loading more state it will be false) {
                mHandler.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        mRefreshLayout.refreshComplete();
                    }
                }, 2000);
            }
        });
```

**Comment by farzadi** [comment]
Thank you

**Comment by farzadi** [comment]
If i don't use `refreshLayout.setHeaderView(new ClassicHeader(getContext())); `app stoping . Is it necessary to use it?

**Comment by dkzwm** [comment]
Can you get the error log to me?
If you wanna use pull down to refresh you can set 
```
mRefreshLayout.setDisableRefresh(false);
mRefreshLayout.setHeaderView(new ClassicHeader(getContext()));
```
otherwise to disable pull down to refresh use code 
```
mRefreshLayout.setDisableRefresh(true);
```
If you wanna use `ClassicHeader` to pull up to refresh, you can use `ClassicFooter`

**Comment by dkzwm** [comment]
You can clone source code,see the demo app source code.
I have a lot of examples in the demo,it will be help you to use this library.

**Comment by farzadi** [comment]
I use it in a fragment. here is my code 

```
    refreshLayout = (SmoothRefreshLayout) rootView.findViewById(R.id.smoothRefreshLayout);
        refreshLayout.setDisableLoadMore(false);
        MaterialFooter footer = new MaterialFooter(getContext());
        footer.setProgressBarColors(new int[]{COLOR_1, COLOR_2, COLOR_3, COLOR_4});
        refreshLayout.setFooterView(footer);
        refreshLayout.setDisableRefresh(true);
        //refreshLayout.setHeaderView(new ClassicHeader(getContext()));
        refreshLayout.setOnRefreshListener(new RefreshingListenerAdapter() {
            @Override
            public void onRefreshBegin(boolean isRefresh) {
...
            }
        });
```


>  java.lang.NullPointerException: Attempt to invoke interface method 'int me.dkzwm.widget.srl.extra.IRefreshView.getStyle()' on a null object reference
>                                                                     at me.dkzwm.widget.srl.SmoothRefreshLayout.layoutFooterView(SmoothRefreshLayout.java:625)
>                                                                     at me.dkzwm.widget.srl.SmoothRefreshLayout.onLayout(SmoothRefreshLayout.java:563)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
>                                                                     at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1079)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
>                                                                     at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1079)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
>                                                                     at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1741)
>                                                                     at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1585)
>                                                                     at android.widget.LinearLayout.onLayout(LinearLayout.java:1494)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
>                                                                     at android.support.design.widget.CoordinatorLayout.layoutChild(CoordinatorLayout.java:1167)
>                                                                     at android.support.design.widget.CoordinatorLayout.onLayoutChild(CoordinatorLayout.java:852)
>                                                                     at android.support.design.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:871)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
>                                                                     at android.support.v4.view.ViewPager.onLayout(ViewPager.java:1795)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
>                                                                     at android.support.design.widget.HeaderScrollingViewBehavior.layoutChild(HeaderScrollingViewBehavior.java:131)
>                                                                     at android.support.design.widget.ViewOffsetBehavior.onLayoutChild(ViewOffsetBehavior.java:42)
>                                                                     at android.support.design.widget.AppBarLayout$ScrollingViewBehavior.onLayoutChild(AppBarLayout.java:1391)
>                                                                     at android.support.design.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:870)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
>                                                                     at android.support.v4.widget.DrawerLayout.onLayout(DrawerLayout.java:1193)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
>                                                                     at android.support.design.widget.CoordinatorLayout.layoutChild(CoordinatorLayout.java:1167)
>                                                                     at android.support.design.widget.CoordinatorLayout.onLayoutChild(CoordinatorLayout.java:852)
>                                                                     at android.support.design.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:871)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
>                                                                     at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1079)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
>                                                                     at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
>                                                                     at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
>                                                                     at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1741)
>                                                                     at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1585)
>                                                                     at android.widget.LinearLayout.onLayout(LinearLayout.java:1494)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
>                                                                     at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
>                                                                     at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
>                                                                     at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1741)
>                                                                     at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1585)
>                                                                     at android.widget.LinearLayout.onLayout(LinearLayout.java:1494)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
> 10-24 22:48:41.299 5186-5186/com.farzad.appr E/AndroidRuntime:     at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
>                                                                     at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
>                                                                     at com.android.internal.policy.DecorView.onLayout(DecorView.java:887)
>                                                                     at android.view.View.layout(View.java:18798)
>                                                                     at android.view.ViewGroup.layout(ViewGroup.java:5951)
>                                                                     at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2648)
>                                                                     at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2364)
>                                                                     at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1515)
>                                                                     at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7091)
>                                                                     at android.view.Choreographer$CallbackRecord.run(Choreographer.java:927)
>                                                                     at android.view.Choreographer.doCallbacks(Choreographer.java:702)
>                                                                     at android.view.Choreographer.doFrame(Choreographer.java:638)
>                                                                     at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:913)
>                                                                     at android.os.Handler.handleCallback(Handler.java:751)
>                                                                     at android.os.Handler.dispatchMessage(Handler.java:95)
>                                                                     at android.os.Looper.loop(Looper.java:154)
>                                                                     at android.app.ActivityThread.main(ActivityThread.java:6682)
>                                                                     at java.lang.reflect.Method.invoke(Native Method)
>                                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
>                                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)

**Comment by dkzwm** [comment]
@farzadi  Sorry,I was negligent. When the footer is in the layout, I checked the style of the header, but header is null, so your app has stopping.I will fix it in version 1.5.0.1.Thank you for your feedback
！

**Comment by dkzwm** [comment]
@farzadi You can depend version 1.5.0.1 now.
